dpnp/backend/extensions/ufunc/elementwise_functions/isclose.cpp

```diff
@@ -0,0 +1,415 @@
+
+    for (int i = 0; i < res_nd; ++i) {
+        nelems *= static_cast<std::size_t>(a_shape[i]);
+        shapes_equal = shapes_equal && (a_shape[i] == res_shape[i] &&
+                                        b_shape[i] == res_shape[i]);
+    }
+    if (!shapes_equal) {
+        throw py::value_error("Array shapes are not the same");
+    }
+
+    // if nelems is zero, return
+    if (nelems == 0) {
+        return std::make_pair(sycl::event(), sycl::event());
+    }
+
+    dpctl::tensor::validation::AmpleMemory::throw_if_not_ample(res, nelems);
+
+    const char *a_data = a.get_data();
+    const char *b_data = b.get_data();
+    char *res_data = res.get_data();
+
+    // handle contiguous inputs
+    const bool is_a_c_contig = a.is_c_contiguous();
+    const bool is_a_f_contig = a.is_f_contiguous();
+
+    const bool is_b_c_contig = b.is_c_contiguous();
+    const bool is_b_f_contig = b.is_f_contiguous();
+
+    const bool is_res_c_contig = res.is_c_contiguous();
+    const bool is_res_f_contig = res.is_f_contiguous();
+
+    const bool all_c_contig =
+        (is_a_c_contig && is_b_c_contig && is_res_c_contig);
+    const bool all_f_contig =
+        (is_a_f_contig && is_b_f_contig && is_res_f_contig);
+
+    if (all_c_contig || all_f_contig) {
+        auto contig_fn = isclose_contig_dispatch_vector[a_b_typeid];
+
+        if (contig_fn == nullptr) {
+            py::dtype a_b_dtype_py = dtype_from_typenum(a_b_typeid);
+            throw std::runtime_error(
+                "Contiguous implementation is missing for " +
+                std::string(py::str(a_b_dtype_py)) + "data type");
+        }
+
+        auto comp_ev = contig_fn(exec_q, nelems, py_rtol, py_atol, py_equal_nan,
+                                 a_data, b_data, res_data, depends);
+        sycl::event ht_ev =
+            dpctl::utils::keep_args_alive(exec_q, {a, b, res}, {comp_ev});
+
+        return std::make_pair(ht_ev, comp_ev);
+    }
+
+    // simplify iteration space
+    //     if 1d with strides 1 - input is contig
+    //     dispatch to strided
+
+    auto const &a_strides = a.get_strides_vector();
+    auto const &b_strides = b.get_strides_vector();
+    auto const &res_strides = res.get_strides_vector();
+
+    using shT = std::vector<py::ssize_t>;
+    shT simplified_shape;
+    shT simplified_a_strides;
+    shT simplified_b_strides;
+    shT simplified_res_strides;
+    py::ssize_t a_offset(0);
+    py::ssize_t b_offset(0);
+    py::ssize_t res_offset(0);
+
+    int nd = res_nd;
+    const py::ssize_t *shape = a_shape;
+
+    py_internal::simplify_iteration_space_3(
+        nd, shape, a_strides, b_strides, res_strides,
+        // output
+        simplified_shape, simplified_a_strides, simplified_b_strides,
+        simplified_res_strides, a_offset, b_offset, res_offset);
+
+    if (nd == 1 && simplified_a_strides[0] == 1 &&
+        simplified_b_strides[0] == 1 && simplified_res_strides[0] == 1)
+    {
+        // Special case of contiguous data
```

Comment:
seems no test is covering that

Reply:
This case is used when we keep the order of the output array as the input if `order=K` using `_empty_like_pair_orderK()` to initialize the output array. This allows us to make the strides of the output array the same as the input and use this special case.
In accordance with NumPy behavior the output array is `always C-contig` regardless of the input data and this part of the code is impossible to use
I have removed it